### PR TITLE
MySQL compatibility: Implement `SHOW COLUMNS`

### LINKED ISSUE #46437: MySQL compatibility: add support for SHOW COLUMNS, SHOW FULL TABLES
**Use case**

ClickHouse with [Looker](https://cloud.google.com/looker) via MySQL protocol.

**Describe the solution you'd like**

When Looker scans a table 

> SHOW FULL COLUMNS FROM `hits_v1` FROM `datasets` LIKE '%'

the operation fails with 
>  <Error> executeQuery: Code: 62. DB::Exception: Syntax error: failed at position 11 ('COLUMNS'): COLUMNS FROM `hits_v1` FROM `datasets` LIKE '%'. Expected one of: DATABASES, CLUSTERS, 

The scan table operation must succeed.

**Additional context**
https://github.com/ClickHouse/ClickHouse/issues/9336#issuecomment-1001090069 
reports `SHOW FULL TABLES FROM .*` failed as well. I confirm the problem exists in `2023.1` release. Maybe we can fix both incompatibility problems. 

cc @alexey-milovidov

### PR DESCRIPTION
**Pull request by rschu1ze**
Fixes #46437

Inspired by https://dev.mysql.com/doc/refman/8.0/en/show-columns.html

### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add statement `SHOW COLUMNS` which shows distilled information from system.columns

### CODE REVIEW

File: tests/queries/0_stateless/25402_show_columns.reference [hunk 1]
```diff
@@ -0,0 +1,38 @@
+int32	Nullable(Int32)	1		\N	
+str	String	0	SOR	\N	
+uint64	UInt64	0	PRI SOR	\N
```
Comment (slvrtrn) [inline tests/queries/0_stateless/25402_show_columns.reference:3]: Please correct me if I am wrong: does it mean that SHOW COLUMNS via MySQL wire protocol will return ClickHouse types in the response without translation to MySQL types?

Comment (rschu1ze) [reply tests/queries/0_stateless/25402_show_columns.reference:3]: To start with some nitpicking (sorry :-) ), let's not confuse the "MySQL wire protocol" with "MySQL SQL dialect". This PR is strictly about the latter.

Yes, we'll return ClickHouse types. Is that a problem? Afaik, no administrative SQL statement in ClickHouse returns the types of another DBMS (and there are many of them: MySQL, Postgres, sqlite, ...).

Comment (slvrtrn) [reply tests/queries/0_stateless/25402_show_columns.reference:3]: @rschu1ze, thanks for the correction :) 
We will check how it behaves with tools such as Looker Studio, where they discover the database schema using the `SHOW COLUMNS` command.

Comment (slvrtrn) [reply tests/queries/0_stateless/25402_show_columns.reference:3]: ^ (edited)

Comment (rschu1ze) [reply tests/queries/0_stateless/25402_show_columns.reference:3]: Sounds good. If we can make SHOW COLUMNS even more MySQL like (it already has some few MySQL-specific quirks, see the [docs](https://github.com/ClickHouse/ClickHouse/pull/48017/files#diff-1b1c5879701b0ee7a53f1a0fd2746d2d9a3793fedc0e0ef366ca107ec8116db9)), I'd be happy to implement it.

**Comment by rschu1ze** [comment]
ClickHouse Integration Tests (asan) [1/6]:
- test_keeper_map/test.py::test_create_drop_keeper_map_concurrent: #48180

**Review by vitlibar — APPROVED** [review]

File: tests/queries/0_stateless/25402_show_columns.sql [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+ENGINE = MergeTree
+ORDER BY uint64;
+
+SHOW COLUMNS FROM tab;
+
+SELECT '---';
+
+SHOW COLUMNS FROM tab FROM database_123456789abcde;
+
+SELECT '---';
```
Comment (vitlibar) [inline tests/queries/0_stateless/25402_show_columns.sql:76]: Something like `SELECT '--a--'`, `SELECT '--b--'` here can make debugging easier :)

Comment (rschu1ze) [reply tests/queries/0_stateless/25402_show_columns.sql:76]: Nice trick, let me try that.

**Pull request merged**